File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string

if len(sys.argv) != 2:
    print("Usage: ./vigenere k")
    exit(1)

# get cipher word
key = sys.argv[1]
base = ""

# transfer cipher word to the base
for i in range(len(key)):
    if key[i].isupper():
        base = base + chr(ord(key[i]) - 64)
    elif key[i].islower():
        base = base + chr(ord(key[i]) - 96)
    else:
        print("Usage: ./vigenere k")
        exit(1)

p_str = get_string("plaintext: ")
cipher_text = ""
# A:65 Z:90 a:97 z:122
counter = 0
for j in range(len(p_str)):
    if p_str[j].isupper():
        tmp = chr((ord(p_str[j]) - 65 + ord(base[counter % len(key)]) - 1) % 26 + 65)
    elif p_str[j].islower():
        tmp = chr((ord(p_str[j]) - 97 + ord(base[counter % len(key)]) - 1) % 26 + 97)
    else:
        tmp = p_str[j]
        counter = counter - 1
    cipher_text = cipher_text + tmp
    counter = counter + 1
print(f"ciphertext: {cipher_text}")

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(int argc, char *argv[])
{
    if (argc == 2)
    {
        /*k is for cipher word*/
        string k = argv[1];
        printf("%s\n", k);
        /*transfer k to the base*/
        for (int i = 0; i < strlen(k); i++)
        {
            if (isupper(k[i]))
            {
                k[i] = k[i] - 64;
            }
            else if(islower(k[i]))
            {
                k[i] = k[i] - 96;
            }
            else
            {
                 printf("Usage: ./vigenere k");
                 return 1;
            }
        }

        printf("plaintext: ");
        string p_text = get_string();

        /* A:65 Z:90 a:97 z:122*/
        int counter = 0;
        for (int i = 0; i < strlen(p_text); i++)
        {
            if (isupper(p_text[i]))
            {
                p_text[i] = (p_text[i] - 65 + k[counter % strlen(k)] - 1) % 26 + 65;
            }
            else if (islower(p_text[i]))
            {
                p_text[i] = (p_text[i] - 97 + k[counter % strlen(k)] - 1) % 26 + 97;
            }
            else
            {
                counter--;
            }
            counter++;
        }
        printf("ciphertext: %s\n", p_text);
        return 0;
    }
    else
    {
        printf("Usage: ./vigenere k");
        return 1;
    }
}
